src/api/legacy.rs

```diff
@@ -327,7 +328,26 @@ async fn read_os_release() -> std::io::Result<HashMap<String, String>> {
 }
 
 async fn read_hostname() -> io::Result<String> {
-    tokio::fs::read_to_string("/proc/sys/kernel/hostname").await
+    let hostname = tokio::fs::read_to_string("/proc/sys/kernel/hostname")
+        .await?
+        .trim_end_matches(|c| c == '\0' || c == '\n')
+        .to_string();
+
+    Ok(hostname)
+}
+
+async fn read_board_model() -> io::Result<(String, String)> {
+    let raw_model = tokio::fs::read_to_string("/proc/device-tree/model")
+        .await?
+        .trim_end_matches(|c| c == '\0' || c == '\n')
+        .to_string();
+
+    let (board_model, board_revision) = raw_model
```

Comment:
This is just an FYI that the model string is not constructed by any convention. We might unknowingly change it, and then this will break. We can leave it for now. We will soon have EEPROM data available which more accurately can tell the version

Reply:
It makes sense, it will never be safe to assume a string will always arrive in a specific shape/form for parsing. At least it would be just refactoring `read_board_model` once EEPROM data is available 😄 

If it helps you, I'll take notes on the EEPROM implementation to validate this afterward!